### DISCUSSION TITLE
Symbol in home page in case of any new message

### DISCUSSION THREAD

**Discussion by guillezcurra** [tomtau/superbasic-im#2] [Ideas]
Maybe a little indicator can be added next to "Chats" in order to check if there is any new message, avoiding having to load all chats to check.
I don’t know if this feature would slow down the website load.

**Comment by tomtau** [comment]
Loading all chats is a bit slow in the beginning.
For this indicator feature, maybe it could set a flag when a new message is received: https://github.com/tomtau/superbasic-im/blob/main/src/client.ts#L35 and reset the flag after "Chats" is opened?

**Reply by guillezcurra** [reply]
Yes, this can be very useful, but is it possible to have an endpoint that returns only the amount of unread messages? This value could be loaded when entering the "new.html" page next to Chats, with this feature there is no need for a flag reset, only checking the amount of unread messages every time you go to "new.html": https://github.com/tomtau/superbasic-im/blob/bca3f7217e263e1675ccf3661a1b6ccefa39f1e7/views/new.html#L17.

If this isn't possible your idea seems great too, and maybe easier to implement.

Thank you!

**Reply by guillezcurra** [reply]
I have been searching whatsapp-web.js info and I found this value"UNREAD_COUNT". Maybe this is what we need?
More info:
https://github.com/pedroslopez/whatsapp-web.js/pull/2059

**Reply by tomtau** [reply]
@guillezcurra I assume that event is fired both for when unread count is increased or reduced? but yes, that could be an option (e.g. have a map/set of unread chats which gets updated in this event listener and the symbol is displayed if this map/set is non-empty)

**Reply by guillezcurra** [reply]
I think your aproach is the best we can get.

I am working on this, the idea is to create an event listener for unread_count and also a variable in client.ts that can be accessed by routes.ts, this variable will be a map <string,integer> that represents <id,unread_count> of each chat, and it is updated in the event listener.
Then, routes.ts will pass the total amount of unread_count in the map to new.html through new_chat_or_pair_handler and the number will be displayed in the "main" page.

What do you think?

Thanks!

**Reply by tomtau** [reply]
@guillezcurra sounds plausible, I opened this issue: https://github.com/tomtau/superbasic-im/issues/10
